runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/stream/KafkaClientConnectionPool.java

```diff
@@ -1306,7 +1380,16 @@ private void onConnectionAbort(
         private void cleanupStreams(
             long traceId)
         {
-            streams.forEach(s -> streamsByInitialIds.remove(s).cleanup(traceId));
+            requests.clear();
+            responses.clear();
+            responseAcks.clear();
+            responseRsts.clear();
+            streams.forEach(s ->
+            {
+                KafkaClientStream stream = streamsByInitialIds.get(s);
+                stream.cleanup(traceId);
+                streamsByInitialIds.remove(s);
```

Comment:
Do we need to remove this one by one, or perhaps this instead?

```
...
streams.forEach(s -> streamsByInitialIds.get(s).cleanup(traceId));
streamsByInitialIds.clear();
...
```

Please rename `streamsByInitialIds` to `streamsByInitialId`.